src/components/Books.js

```diff
@@ -0,0 +1,76 @@
+import React, { useState } from 'react';
+import { useDispatch } from 'react-redux';
+import { addBookAsync } from '../redux/books/booksSlice';
+import BookList from './BooksList';
+import Header from './Header';
+import './Main.css';
+
+// end imports
+
+const App = () => {
+  const dispatch = useDispatch();
+  const [selectedApp] = useState('bMrmeF1Ldn440EWBDcT5');
+
+  const [book, setBook] = useState({
+    title: '',
+    author: '',
+  });
+
+  const handleChange = (e) => {
+    const { name, value } = e.target;
+    setBook({ ...book, [name]: value });
+  };
+
+  function generateUUID() {
+    return 'xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx'.replace(/[xy]/g, (c) => {
+      const r = Math.random() * 16 | 0; const
+        v = c === 'x' ? r : (r & 0x3 | 0x8);
+      return v.toString(16);
+    });
+  }
+
+  const handleAddBook = async () => {
+    const newBook = {
+      item_id: generateUUID(),
+      title: book.title,
+      author: book.author,
+      category: 'Fiction',
+    };
+
+    try {
+      await dispatch(addBookAsync({ app_id: selectedApp, newBook }));
```

Comment:
- [ ] After adding a book the title and the author details are not displayed on the page until the page is refreshed. I think this is an aspect of your app that can be improved for a better experience. The state should be correctly updated.

  Book is added, but not displayed correctly | Displayed correctly only after refresh
  :---:|:---:
  ![imgepe](https://github.com/paulshaun-kdtk/React_Book_Store/assets/54848363/f168d305-5c50-4cd6-85cd-410b2ec8f1f5) | ![imgepe-2](https://github.com/paulshaun-kdtk/React_Book_Store/assets/54848363/6cbbc526-0592-41df-b21b-8493050bef61)